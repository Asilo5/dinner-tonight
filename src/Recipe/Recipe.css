.recipe-img {
  height: 15em;
  margin: 10px;
}

.recipe {
  border: 4px #ff3838 solid;
  padding: 10px;
  border-radius: 20px;
  margin: 20px;
}

h3 {
  font-family: 'Gayathri', sans-serif;
  font-size: 27px;
  margin: 10px;
}

p {
  font-family: 'Gayathri', sans-serif; 
}