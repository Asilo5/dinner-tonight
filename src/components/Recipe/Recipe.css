.recipe-img {
  height: 15em;
  margin: 10px;
}

.recipe {
  border: 4px #ff3838 solid;
  padding: 10px;
  border-radius: 20px;
  margin: 20px;
}

h3 {
  font-family: 'Gayathri', sans-serif;
  font-size: 27px;
  margin: 10px;
}

p {
  font-family: 'Gayathri', sans-serif; 
}

.view-recipe-btn {
    padding: 10px;
    font-size: 20px;
    border: 2px solid #ff3838;
    margin: 10px;
    border-radius: 5px;
    width: 50%;
    background-color: transparent;
}

.view-recipe-btn:hover {
  background: #ff3838;
  color: white;
}

a {
  text-decoration: none;
  color: black;
}

a:hover{
    text-decoration: underline;
    color: white;
    cursor: grabbing;
  }
