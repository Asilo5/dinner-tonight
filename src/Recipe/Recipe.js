import React from 'react';
import './Recipe.css'

const Recipe = ({title, publisher, image, recipeLink}) => {
    return(
        <section className='recipe'>
          <h3> {title} </h3>
          <img className='recipe-img' src={image} alt={title}/>
          <p> Publisher: {publisher}</p>
          <button className='view-recipe-btn'><a href={recipeLink}>View Recipe</a></button>
        </section>
    )
}

export default Recipe;