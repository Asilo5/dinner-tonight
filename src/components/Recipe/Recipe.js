import React from 'react';
import './Recipe.css';
import PropTypes from 'prop-types';

const Recipe = ({title, publisher, image, recipeLink}) => {
    return(
        <section className='recipe'>
          <h3> {title} </h3>
          <img className='recipe-img' src={image} alt={title}/>
          <p> By: {publisher}</p>
          <button className='view-recipe-btn'><a href={recipeLink}>View Recipe</a></button>
        </section>
    ) 
}

Recipe.propTypes = {
  title: PropTypes.string, 
  publisher: PropTypes.string, 
  image: PropTypes.string, 
  recipeLink: PropTypes.string
}

export default Recipe;