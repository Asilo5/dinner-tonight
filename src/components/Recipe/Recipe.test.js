import React from 'react';
import { shallow } from 'enzyme';
import Recipe from './Recipe';

describe('Recipe', () => {

    let wrapper;
    

    beforeEach(() => {
        wrapper = shallow(<Recipe id='ed9409'
                                  title='Gluten-Free Vegan Banana Peanut Butter Chocolate Chip Cookies'
                                  publisher='Two Peas and Their Pod'
                                  image='http://static.food2fork.com/BananaPeanutButterChocolateChipCookies13e87.jpg'
                                  recipeLink='http://www.twopeasandtheirpod.com/gluten-free-vegan-banana-peanut-butter-chocolate-chip-cookies/' />)
    })

    it('should match snapshot with all information passing in correctly', () => {
        expect(wrapper).toMatchSnapshot();
    })
})