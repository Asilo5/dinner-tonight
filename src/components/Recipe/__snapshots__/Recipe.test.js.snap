// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Recipe should match snapshot with all information passing in correctly 1`] = `
<section
  className="recipe"
>
  <h3>
     
    Gluten-Free Vegan Banana Peanut Butter Chocolate Chip Cookies
     
  </h3>
  <img
    alt="Gluten-Free Vegan Banana Peanut Butter Chocolate Chip Cookies"
    className="recipe-img"
    src="http://static.food2fork.com/BananaPeanutButterChocolateChipCookies13e87.jpg"
  />
  <p>
     By: 
    Two Peas and Their Pod
  </p>
  <button
    className="view-recipe-btn"
  >
    <a
      href="http://www.twopeasandtheirpod.com/gluten-free-vegan-banana-peanut-butter-chocolate-chip-cookies/"
    >
      View Recipe
    </a>
  </button>
</section>
`;
